tests/test_widget.py

```diff
@@ -1,7 +1,63 @@
+import os
+import time
 from random import random
+from typing import Callable
 
 import numpy as np
 import pytest
+import tifffile as tf
+from qtpy.QtCore import Qt, QTimer
+from qtpy.QtWidgets import (
+    QApplication,
+    QDialog,
+    QDialogButtonBox,
+)
+
+
+def get_dialog(
+    dialog_trigger: Callable,
+    dialog_action: Callable,
+    time_out: int = 5,
+    trigger_option: int = 1,
+) -> QDialog:
+    """
+    Returns the current dialog (active modal widget). If there is no
+    dialog, it waits until one is created for a maximum of 5 seconds (by
+    default).
+
+    :param dialog_trigger: Callable that triggers the dialog creation.
+    :param time_out: Maximum time (seconds) to wait for the dialog creation.
+    """
+
+    dialog: QDialog = None
+    start_time = time.time()
+
+    # Helper function to catch the dialog instance and hide it
+    def dialog_creation():
+        # Wait for the dialog to be created or timeout
+        nonlocal dialog
+        while dialog is None and time.time() - start_time < time_out:
+            dialog = QApplication.activeModalWidget()
+
+        # Avoid errors when dialog is not created
+        if dialog is not None:
+            # Perform action dialog_action()
+            # dialog.hide() and dialog.close() do not seem to work
+            dialog_action(dialog)
+
+    # Create a thread to get the dialog instance and call dialog_creation trigger
+    QTimer.singleShot(1, dialog_creation)
+    dialog_trigger(trigger_option)
```

Comment:
This would be clearer if you called the function `dialog_capture`. You create a thread and wait 1ms before calling dialog capture. And in the main thread (instantly!) you trigger the _creation_ of the dialog with the `dialog_trigger` function.